src/compiler/transformers/declarations/emitBinder.ts

```diff
@@ -0,0 +1,914 @@
+    [SyntaxKind.CallSignature]: { includes: SymbolFlags.Signature, excludes: SymbolFlags.None },
+    [SyntaxKind.ConstructSignature]: { includes: SymbolFlags.Signature, excludes: SymbolFlags.None },
+    [SyntaxKind.IndexSignature]: { includes: SymbolFlags.Signature, excludes: SymbolFlags.None },
+    [SyntaxKind.MethodDeclaration]: { includes: SymbolFlags.Method, excludes: SymbolFlags.MethodExcludes },
+    [SyntaxKind.MethodSignature]: { includes: SymbolFlags.Method, excludes: SymbolFlags.MethodExcludes },
+    [SyntaxKind.FunctionDeclaration]: { includes: SymbolFlags.Function, excludes: SymbolFlags.FunctionExcludes },
+    [SyntaxKind.Constructor]: { includes: SymbolFlags.Constructor, excludes: SymbolFlags.None },
+    [SyntaxKind.GetAccessor]: { includes: SymbolFlags.GetAccessor, excludes: SymbolFlags.GetAccessorExcludes },
+    [SyntaxKind.SetAccessor]: { includes: SymbolFlags.SetAccessor, excludes: SymbolFlags.SetAccessorExcludes },
+    [SyntaxKind.ClassExpression]: { includes: SymbolFlags.Class, excludes: SymbolFlags.ClassExcludes },
+    [SyntaxKind.ClassDeclaration]: { includes: SymbolFlags.Class, excludes: SymbolFlags.ClassExcludes },
+    [SyntaxKind.InterfaceDeclaration]: { includes: SymbolFlags.Interface, excludes: SymbolFlags.InterfaceExcludes },
+    [SyntaxKind.TypeAliasDeclaration]: { includes: SymbolFlags.TypeAlias, excludes: SymbolFlags.TypeAliasExcludes },
+    [SyntaxKind.EnumDeclaration]: {
+        const: { includes: SymbolFlags.ConstEnum, excludes: SymbolFlags.ConstEnumExcludes },
+        regular: { includes: SymbolFlags.RegularEnum, excludes: SymbolFlags.RegularEnumExcludes },
+    },
+    [SyntaxKind.ModuleDeclaration]: {
+        value: { includes: SymbolFlags.ValueModule, excludes: SymbolFlags.ValueModuleExcludes },
+        namespace: { includes: SymbolFlags.NamespaceModule, excludes: SymbolFlags.NamespaceModuleExcludes },
+    },
+    [SyntaxKind.ImportEqualsDeclaration]: { includes: SymbolFlags.Alias, excludes: SymbolFlags.AliasExcludes },
+    [SyntaxKind.NamespaceImport]: { includes: SymbolFlags.Alias, excludes: SymbolFlags.AliasExcludes },
+    [SyntaxKind.ImportSpecifier]: { includes: SymbolFlags.Alias, excludes: SymbolFlags.AliasExcludes },
+    [SyntaxKind.ExportSpecifier]: { includes: SymbolFlags.Alias | SymbolFlags.ExportValue, excludes: SymbolFlags.AliasExcludes },
+    [SyntaxKind.NamespaceExportDeclaration]: { includes: SymbolFlags.Alias, excludes: SymbolFlags.AliasExcludes },
+    [SyntaxKind.ImportClause]: { includes: SymbolFlags.Alias, excludes: SymbolFlags.AliasExcludes },
+} as const satisfies Partial<Record<SyntaxKind, SymbolRegistrationFlags | Record<string, SymbolRegistrationFlags>>>;
+
+/**
+ * Assigning values to a property of a function will usually cause those members to be implicitly declared on the function
+ * even if they were were not declared (expando functions)
+ * DTE needs to detect these members and error on them since this behavior is not supported in isolated declarations

```

Comment:
The term ***DTE*** was defined as a "`.d.ts`-emitter" by @RyanCavanaugh in [this post](https://github.com/microsoft/TypeScript/issues/53500#issuecomment-1485740788) from March 2023 and has been used in the context of this project semi-frequently since then.

It refers to a tool that can take exactly one `.ts` file and emit exactly one `.d.ts` file as a simple syntactic transform. It is expected that all future DTEs will match the reference implementation in TypeScript.